assets/src/vo/money.js

```diff
@@ -0,0 +1,519 @@
+/**
+ * External imports
+ */
+import { Decimal } from 'decimal.js-light';
+import * as Accounting from 'accounting-js';
+import isShallowEqual from '@wordpress/is-shallow-equal';
+import { Exception } from '@eventespresso/eejs';
+import { isEmpty } from 'lodash';
+
+/**
+ * Internal imports
+ */
+import { Currency } from './currency';
+
+/**
+ * Asserts if incoming value is an instance of Money
+ * @param {Money} money
+ * @throws {TypeError}
+ */
+const assertMoney = ( money ) => {
+	if ( ! ( money instanceof Money ) ) {
+		throw new TypeError( 'Instance of Money required' );
+	}
+};
+
+/**
+ * Asserts if incoming value is an instance of Currency
+ * @param {Currency} currency
+ * @throws {TypeError}
+ */
+const assertCurrency = ( currency ) => {
+	if ( ! ( currency instanceof Currency ) ) {
+		throw new TypeError( 'Instance of Currency required' );
+	}
+};
+
+/**
+ * Asserts if two currencies are shallow equal.
+ * @param {Currency} currencyA
+ * @param {Currency} currencyB
+ * @throws {Exception}
+ */
+const assertEqualCurrency = ( currencyA, currencyB ) => {
```

Comment:
The reason I didn't go with "same" is because I thought that may get confused with it asserting that the two values are the same _instance_.  It doesn't, it asserts if the values represented by the instances are _equal_.  There's a subtle but **important** distinction there.